holoviews/core/data/__init__.py

```diff
@@ -284,7 +284,7 @@ def range(self, dim, data_range=True, dimension_range=True):
             return (None, None)
         elif all(util.isfinite(v) for v in dim.range) and dimension_range:
             return dim.range
-        elif dim in self.dimensions() and data_range and len(self):
+        elif dim in self.dimensions() and data_range and bool(self):
```

Comment:
This is mainly an optimization, computing the length on a Polygons/Path can be very expensive, ``bool(self)`` is equivalent but a much quicker way to check whether the element is empty.